graphql_compiler/compiler/ir_lowering_cypher/ir_lowering.py

```diff
@@ -89,16 +92,89 @@ def remove_mark_location_after_optional_backtrack(ir_blocks, query_metadata_tabl
     return new_ir_blocks
 
 
+class CypherFoldScopeContextField(Expression):
+    """An expression for a field captured in a @fold scope for non _x_count filtering."""
```

Comment:
I think a few updates and more bits of information would be nice here.
- I think `non _x_count filtering` is too restrictive as a description here. `_x_count` seems disallowed altogether in the `validate()` function, so it's not just about filtering on it.
- I'd explain how this is different from a regular `FoldedContextField`. Why do we need a separate class for it?
- Bonus points if you come up with a name that suggests the above difference immediately, so that readers of the code don't have to reach for the docstring to understand the distinction – especially if it helps avoid the confusion when they notice that both regular `FoldedContextField` objects and these new objects are used simultaneously.

Reply:
updated: the name is still really long but it's clearer that it's for pre-folding work